modules/sb/fsw/src/cfe_sb_buf.c

```diff
@@ -145,14 +145,23 @@ CFE_SB_BufferD_t *CFE_SB_GetBufferFromPool(size_t MaxMsgSize)
  *-----------------------------------------------------------------*/
 void CFE_SB_ReturnBufferToPool(CFE_SB_BufferD_t *bd)
 {
+    int32 Status;
+
     /* Remove from any tracking list (no effect if not in a list) */
     CFE_SB_TrackingListRemove(&bd->Link);
 
     --CFE_SB_Global.StatTlmMsg.Payload.SBBuffersInUse;
     CFE_SB_Global.StatTlmMsg.Payload.MemInUse -= bd->AllocatedSize;
 
     /* finally give the buf descriptor back to the buf descriptor pool */
-    CFE_ES_PutPoolBuf(CFE_SB_Global.Mem.PoolHdl, bd);
+    Status = CFE_ES_PutPoolBuf(CFE_SB_Global.Mem.PoolHdl, bd);
+
+    if (Status < 0)
+    {
+        CFE_ES_WriteToSysLog("%s: PutPoolBuf Fail Stat=0x%08X, Hndl=0x%08lX, Buf=0x%08lX\n", __func__,
```

Comment:
Consider only writing once to avoid flooding the log.

Reply:
Jake @skliper, how would I go about limiting it to one write?
Cheers

Reply:
Relates to:
-  #1469

It's a bigger conceptual change, but I'm a big fan of persistent indication of system health/issues of this flavor.  So if there was a bitfield/counter or similar in HK to track these sort of issues, you could only print if there wasn't a previous failure of this type indicated.